### ISSUE TITLE
Tf.train.Saver() Failing with V2 Checkpoint

### ISSUE THREAD

**Issue by danleh93** [tensorflow/models#724]
There is a new way of saving checkpoint files ending in a .data-00000-of-00001 format.

I am trying to restore this file via:

```
sess = tf.Session()
saver = tf.train.Saver(write_version=tf.train.SaverDef.V2)
saver.restore(sess, '/home/foo/bar/model.ckpt-19462.data-00000-of-00001')
```

However, I get the following error:

> DataLossError (see above for traceback): Unable to open table file /home/foo/bar/model.ckpt-19462.data-00000-of-00001: Data loss: not an sstable (bad magic number): perhaps your file is in a different file format and you need to use a different restore operator?

Stack Trace:

```
Caused by op u'save/RestoreV2_221', defined at:
  File "classify_image.py", line 95, in <module>
    saver = tf.train.Saver(write_version=2)
  File "/usr/local/lib/python2.7/dist-packages/tensorflow/python/training/saver.py", line 1028, in __init__
    self.build()
  File "/usr/local/lib/python2.7/dist-packages/tensorflow/python/training/saver.py", line 1058, in build
    restore_sequentially=self._restore_sequentially)
  File "/usr/local/lib/python2.7/dist-packages/tensorflow/python/training/saver.py", line 652, in build
    restore_sequentially, reshape)
  File "/usr/local/lib/python2.7/dist-packages/tensorflow/python/training/saver.py", line 386, in _AddRestoreOps
    tensors = self.restore_op(filename_tensor, saveable, preferred_shard)
  File "/usr/local/lib/python2.7/dist-packages/tensorflow/python/training/saver.py", line 225, in restore_op
    [spec.tensor.dtype])[0])
  File "/usr/local/lib/python2.7/dist-packages/tensorflow/python/ops/gen_io_ops.py", line 441, in restore_v2
    dtypes=dtypes, name=name)
  File "/usr/local/lib/python2.7/dist-packages/tensorflow/python/framework/op_def_library.py", line 763, in apply_op
    op_def=op_def)
  File "/usr/local/lib/python2.7/dist-packages/tensorflow/python/framework/ops.py", line 2371, in create_op
    original_op=self._default_original_op, op_def=op_def)
  File "/usr/local/lib/python2.7/dist-packages/tensorflow/python/framework/ops.py", line 1258, in __init__
    self._traceback = _extract_stack()
```

Even if I use `saver = tf.train.Saver()` without specifying the write version, I still get the same error.
